forte/data/ontology/ontology_code_generator.py

```diff
@@ -454,38 +461,82 @@ def parse_ontology_spec(self, json_file_path: str,
+
+        for user_import in json_imports:
+            installed_json_file = None
+            is_package = '.' in user_import and '/' not in user_import
+            if is_package:
+                # Check installed modules
+                try:
+                    import_origin = importlib.util.find_spec(user_import).origin
+
+                    delimiters = '# ', SOURCE_JSON_PFX, SOURCE_JSON_SFX
+                    regex = '|'.join(map(re.escape, delimiters))
+                    reqd_line = open(import_origin, 'r').readlines()[1]
+                    installed_json_file = list(filter(
+                        None, re.split(regex, reqd_line)))[0]
+
+                    schema_to_import = installed_json_file
+
+                    self.json_paths.append(installed_forte_dir)
+                except (AttributeError, IndexError, FileNotFoundError) as err:
+                    raise ImportOntologyNotFoundException(
+                        f"Ontology corresponding to package {user_import} not "
+                        f"found to be installed.")
+            else:
+                schema_to_import = user_import
+            # Check for installed ontology source file or json_file in user
+            # provided directories
+            import_json_file = utils.search_in_dirs(schema_to_import,
                                                     self.json_paths)
             if import_json_file is None:
                 raise ImportOntologyNotFoundException(
-                    f"Ontology corresponding to {import_file} not "
+                    f"Ontology corresponding to {user_import} not "
                     f"found in the current directory or the "
-                    f"directory of original json config.")
+                    f"directory of source json config.")
+
             if import_json_file in rec_visited_paths:
                 raise ImportOntologyAlreadyGeneratedException(
                     f"Ontology corresponding to {import_json_file}"
                     f" already generated, cycles not permitted, "
                     f"aborting")
             elif import_json_file not in visited_paths:
                 self.parse_ontology_spec(
-                    import_json_file, destination_dir,
+                    import_json_file, destination_dir, installed_json_file,
                     visited_paths, rec_visited_paths)
 
         # Once the ontology for all the imported files is generated, generate
         # ontology of the current file.
-        self.parse_schema(spec_dict)
+
+        is_installed = True
+        if forte_json_file is None:
+            is_installed = False
+            forte_json_file = json_file_path
+
+        curr_forte_dir = str(Path(os.path.join(os.path.dirname(__file__),
+                                               *([os.pardir] * 3))).resolve())
+
+        if os.path.samefile(curr_forte_dir, installed_forte_dir):
+            forte_json_file = os.path.relpath(forte_json_file, curr_forte_dir)
```

Comment:
Why convert to the relative path here? Can we directly use `forte_json_file` without this step?

Reply:
Let's say one of the forte developers runs the generation, we would not want to write the absolute path in the comments. A relative path essentially means that the source json file lies in the same directory as the generated python ontology. This is important when we read the path again.